### PR TITLE
fix: fix two minor issues of bulti-in authn/authz

### PR DESCRIPTION
**Pull request by lafirest**
1. the `Derived Key Length` for `pbkdf2` should be a positive integer.
2. fix topics in the authorization rules might be parsed incorrectly

Fixes EMQX-11414 EMQX-10583

Release version: v/e5.?

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_auth/src/emqx_authz/emqx_authz_rule_raw.erl [hunk 1]
```diff
@@ -49,6 +49,7 @@ parse_rule(
         Permission = validate_rule_permission(PermissionRaw),
         ActionType = validate_rule_action_type(ActionTypeRaw),
         Action = validate_rule_action(ActionType, RuleRaw),
+        io:format(">>> Topics:~p~n", [Topics]),
```
Comment (zmstone) [inline apps/emqx_auth/src/emqx_authz/emqx_authz_rule_raw.erl:52]: ？？

File: apps/emqx_auth/src/emqx_authz/emqx_authz_rule_raw.erl [hunk 2]
```diff
@@ -98,7 +99,7 @@ validate_rule_topics(RuleRaw) ->
     throw({missing_topic_or_topics, RuleRaw}).
 
 validate_rule_topic(<<"eq ", TopicRaw/binary>>) ->
-    {eq, validate_rule_topic(TopicRaw)};
```
Comment (zmstone) [inline apps/emqx_auth/src/emqx_authz/emqx_authz_rule_raw.erl:101]: meaning?
`<<"eq eq a/#">>` is parsed as `{eq, <<"eq a/#">>}`?

Comment (lafirest) [reply apps/emqx_auth/src/emqx_authz/emqx_authz_rule_raw.erl:101]: Yes, and "eq a/#" is a valid topic

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2880)

**Pull request merged**